### PR TITLE
Added documentation

### PR DESCRIPTION
**Pull request by Superdoude1999**
Added most of missing documentation (didn't do utils folder).
Also changed the title of the sold item list fragment.

### CODE REVIEW

**Review by v1ruso — APPROVED** [review]

File: app/src/main/java/com/example/sharingang/AbstractFirestoreStore.kt [hunk 1]
```diff
@@ -57,6 +81,11 @@ abstract class AbstractFirestoreStore<T : Any>(
         }
     }
 
+    /**
+     * Function to remove an element from the database.
+     * @param[id] The id of the element to remove.
```
Comment (lumapools) [inline app/src/main/java/com/example/sharingang/AbstractFirestoreStore.kt:86]:
```suggestion
     * @param id The id of the element to remove.
```

File: app/src/main/java/com/example/sharingang/DateConverter.kt [hunk 2]
```diff
@@ -7,6 +7,12 @@ import java.util.*
  * A utility object needed to store Date fields in our Room entities.
  */
 object DateConverter {
+
+    /**
+     * Function to convert timestamps (Long) to Date.
+     * @param[timestamp]
```
Comment (lumapools) [inline app/src/main/java/com/example/sharingang/DateConverter.kt:13]:
```suggestion
     * @param timestamp
```

File: app/src/main/java/com/example/sharingang/ListConverter.kt [hunk 3]
```diff
@@ -12,6 +21,12 @@ object ListConverter {
         return stringList.split(',')
     }
 
+    /**
+     * Function to convert List of Strings into single String.
+     * Uses commas to separate elements.
+     * @param[list] List of Strings to convert.
```
Comment (lumapools) [inline app/src/main/java/com/example/sharingang/ListConverter.kt:27]:
```suggestion
     * @param list List of Strings to convert.
```

File: app/src/main/java/com/example/sharingang/items/ItemDao.kt [hunk 4]
```diff
@@ -3,23 +3,47 @@ package com.example.sharingang.items
...
+     * Function to get one item from database.
+     * @param[id] The id of the item to get.
+     * @return Item corresponding to the id.
+     */
     @Query("SELECT * FROM item WHERE id = :id")
     suspend fun getItem(id: String): Item?
 
+    /**
+     * Function to get all items posted by a user.
+     * @param[userId] The id of the user.
```
Comment (lumapools) [inline app/src/main/java/com/example/sharingang/items/ItemDao.kt:28]:
```suggestion
     * @param userId The id of the user.
```

File: app/src/main/java/com/example/sharingang/items/ItemDao.kt [hunk 5]
```diff
@@ -3,23 +3,47 @@ package com.example.sharingang.items
...
 
+    /**
+     * Function to remove all items from database.
+     */
     @Query("DELETE FROM item")
     fun clear()
 
+    /**
+     * Function to replace all items in database with a new list of items.
+     * @param[items] List aof new items.
```
Comment (lumapools) [inline app/src/main/java/com/example/sharingang/items/ItemDao.kt:45]:
```suggestion
     * @param items List of new items.
```

File: app/src/main/java/com/example/sharingang/users/UserDao.kt [hunk 6]
```diff
@@ -3,20 +3,45 @@ package com.example.sharingang.users
...
+     * Retrieve a user from database.
+     * @param[id] the id of the user.
+     * @return The corresponding User.
+     */
     @Query("SELECT * FROM user WHERE id = :id")
     fun getUser(id: String): User?
 
+    /**
+     * Add a list of users to the database.
+     * @param[users] List of users to add.
```
Comment (lumapools) [inline app/src/main/java/com/example/sharingang/users/UserDao.kt:30]:
```suggestion
     * @param users List of users to add.
```

File: app/src/main/java/com/example/sharingang/users/UserDao.kt [hunk 7]
```diff
@@ -3,20 +3,45 @@ package com.example.sharingang.users
...
 
+    /**
+     * Remove all users from database.
+     */
     @Query("DELETE FROM user")
     fun clear()
 
+    /**
+     * Replace all users with provided list.
+     * @param[users] List of users to replace with.
```
Comment (lumapools) [inline app/src/main/java/com/example/sharingang/users/UserDao.kt:43]:
```suggestion
     * @param users List of users to replace with.
```

**Review by lumapools — CHANGES_REQUESTED** [review]
Thank you for documenting! Please could you see my requested changes? I don't think you need `[ ]` around `@param`s.
Also, instead of writing `Function that does this and that` for the documentation, it is better practice to write `Does this and that`. Same for classes , i.e. don't specify `Class that represents ...` but just `Represents ...`. Also, make sure to add `@property` for the fields contained by a Class.

File: app/src/main/java/com/example/sharingang/AbstractFirestoreStore.kt [hunk 8]
```diff
@@ -4,13 +4,22 @@ import android.util.Log
 import com.google.firebase.firestore.FirebaseFirestore
 import kotlinx.coroutines.tasks.await
 
+/**
+ * Abstract class to implement basic database usage.
+ * @param[T] Type of element to use for database.
+ */
```
Comment (lumapools) [inline app/src/main/java/com/example/sharingang/AbstractFirestoreStore.kt:10]:
```suggestion
/**
 * Abstract class to implement basic database usage.
 * @param[T] Type of element to use for database.
 * @property collectionName The name of the collection
 * @property typeClass Class with type T
 * @property firestore instance of FirebaseFirestore
 */
```

File: app/src/main/java/com/example/sharingang/items/CachedItemRepository.kt [hunk 9]
```diff
@@ -6,6 +6,9 @@ import kotlinx.coroutines.withContext
 import javax.inject.Inject
 
 
+/**
+ * Class to implement an ItemRepository in cache.
+ */
```
Comment (lumapools) [inline app/src/main/java/com/example/sharingang/items/CachedItemRepository.kt:11]:
```suggestion
/**
 * Class to implement an ItemRepository in cache.
 * @property itemDao used item data access object
 * @property store used ItemStore
 */
```

File: app/src/main/java/com/example/sharingang/items/ItemsAdapter.kt [hunk 10]
```diff
@@ -19,6 +22,9 @@ class ItemsAdapter(private val clickListener: ItemListener) :
         holder.bind(getItem(position)!!, clickListener)
     }
 
+    /**
+     * Recycler view to represent a single item.
+     */
```
Comment (lumapools) [inline app/src/main/java/com/example/sharingang/items/ItemsAdapter.kt:27]:
```suggestion
    /**
     * Recycler view to represent a single item.
     * @property binding item view binding
     */
```

File: app/src/main/java/com/example/sharingang/users/FirestoreCurrentUserProvider.kt [hunk 11]
```diff
@@ -3,6 +3,9 @@ package com.example.sharingang.users
 import com.google.firebase.auth.FirebaseAuth
 import javax.inject.Inject
 
+/**
+ * Class used to retrieve the current signed in user.
+ */
```
Comment (lumapools) [inline app/src/main/java/com/example/sharingang/users/FirestoreCurrentUserProvider.kt:8]:
```suggestion
/**
 * Class used to retrieve the current signed in user.
 * @property auth the authenticator to use
 */
```

**Review by lumapools — APPROVED** [review]
Added some suggestions for some missed `@property`s. Good work overall!

**Pull request merged**